pages/answers.tsx

```diff
@@ -1,44 +1,147 @@
-  //.transform((value) => new Date(value)),
   public_key: z.string(),
   sealed_secret: z.string(),
   signature: z.string(),
 });
 
-const jsonDataSchema = z.object({
-  firstName: z.string(),
-  lastName: z.string(),
-  message: z.string(),
-});
-
-const jsonValidationSchema = formMetadata.extend(jsonDataSchema.shape);
-
 type EncryptedAnswer = z.infer<typeof formMetadata> & { data: string };
 
-type DecryptedData = z.infer<typeof jsonDataSchema>;
-
-type DecryptedAnswer = z.infer<typeof jsonValidationSchema>;
-
 type AnswersResponse = {
   errors?: { message: string }[];
   data: {
     answers: EncryptedAnswer[];
   };
 };
 
+const columns: GridColDef[] = [
+  { field: "id", headerName: "ID", width: 70 },
+  {
+    field: "mood",
+    headerName: "Mood",
+    description: "This column has a value getter and is not sortable.",
+    sortable: false,
+    align: "center",
+    width: 40,
+    renderCell: (cell) =>
+      cell.row.color && (
+        <Avatar
+          size={30}
+          name={cell.row.color}
+          variant="pixel"
+          colors={["#92A1C6", "#146A7C", "#F0AB3D", "#C271B4", "#C20D90"]}
+        />
+      ),
+  },
+  {
+    field: "created_at",
+    headerName: "Created at",
+    width: 130,
+    type: "date",
+    valueGetter: (val) => new Date(val.row.created_at),
+  },
+  {
+    field: "firstName",
+    type: "string",
+    headerName: "First name",
+    width: 120,
+  },
+  {
+    field: "lastName",
+    type: "string",
+    headerName: "Last name",
+    width: 120,
+  },
+  {
+    field: "email",
+    headerName: "Email",
+    type: "email",
+    width: 150,
+    renderCell: (cell) => (
+      <a href={`mailto:${cell.row.email}`}>{cell.row.email}</a>
+    ),
+  },
+  {
+    field: "color",
+    headerName: "Color",
+    width: 150,
+  },
+  {
+    field: "newsletter",
+    headerName: "Emails",
+    width: 70,
+    align: "center",
+    valueGetter: (val) => (val.row.newsletter && "✅") || "❌",
+  },
+  {
+    field: "alerts",
+    headerName: "Alertes",
+    width: 70,
+    align: "center",
+    valueGetter: (val) => (val.row.alerts && "✅") || "❌",
+  },
+  {
+    field: "message",
+    headerName: "Message",
+    type: "text",
+    flex: 1,
+  },
+];
+
+const decryptAnswer = (client: Client, answer: EncryptedAnswer) => {
+  try {
+    const values = client.unsealFormData(
+      {
+        metadata: {
+          publicKey: answer.public_key,
+          sealedSecret: answer.sealed_secret,
+          signature: answer.signature,
+        },
+        encrypted: { data: answer.data },
+      },
+      nameFingerprint
+    ) as Record<"data", string>;
+    const decryptedValues: FormData = JSON.parse(values.data);
+    const res = jsonDataSchema.safeParse(decryptedValues);
+    if (!res.success) {
+      console.error(`Zod: Impossible de parser la réponse ${answer.id}`);
+      // return null here if you dont want to accept invalid zod schemas
+      return {
+        ...answer,
+        ...decryptedValues,
+      };
```

Comment:
Note: returning `null` here is recommended to avoid security issues where malicious content is sent that could break the rendering process, and lead to a blank page.